dali/pipeline/data/tensor.h

```diff
@@ -314,9 +316,10 @@ class Tensor : public Buffer<Backend> {
     data_.reset(tl->raw_mutable_tensor(0), [](void *) {});
 
     // Get the meta-data for the target tensor
-    shape_ = tl->tensor_shape(0);
-    shape_.insert(shape_.begin(), tl->ntensor());
-    size_ = volume(shape_);
+    auto new_shape = tl->tensor_shape(0);
```

Comment:
Why not:
```
auto new_shape = tl->tensor_shape(0);
new_shape.insert(new_shape.begin(), tl->ntensor());
shape_ = kernels::TensorShape<>(new_shape);
size_ = volume(shape_);
```

Reply:
I have it written as shape_cat in follow-up, but for that I need to use TensorListShape in TensorList, so tensor_shape(0) returns me a TensorShape<> and not a vector<int64_t>